src/test/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/KendraIntelligentRankerTests.java

```diff
@@ -0,0 +1,210 @@
+import org.opensearch.search.relevance.transformer.kendraintelligentranking.model.dto.RescoreResult;
+import org.opensearch.search.relevance.transformer.kendraintelligentranking.model.dto.RescoreResultItem;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class KendraIntelligentRankerTests extends OpenSearchTestCase {
+    private static KendraHttpClient buildMockHttpClient(Function<RescoreRequest, RescoreResult> mockRescoreImpl) {
+        KendraHttpClient kendraHttpClient = Mockito.mock(KendraHttpClient.class);
+        Mockito.doAnswer(invocation -> {
+            RescoreRequest rescoreRequest = invocation.getArgument(0);
+            return mockRescoreImpl.apply(rescoreRequest);
+        }).when(kendraHttpClient).rescore(Mockito.any(RescoreRequest.class));
+        return kendraHttpClient;
+    }
+
+    private static KendraHttpClient buildMockHttpClient() {
+        return buildMockHttpClient(r -> new RescoreResult());
+    }
+
+    public void testGetSettings() {
+        List<Setting<?>> settings = new KendraIntelligentRanker(buildMockHttpClient()).getTransformerSettings();
+        assertNotNull(settings);
+        assertFalse(settings.isEmpty());
+    }
+
+    public void testPreprocess() {
+        KendraIntelligentRanker ranker = new KendraIntelligentRanker(buildMockHttpClient());
+        KendraIntelligentRankingProperties properties =
+                new KendraIntelligentRankingProperties(List.of("body"), List.of("title"), 50);
+        ResultTransformerConfiguration configuration = new KendraIntelligentRankingConfiguration(1, properties);
+        SearchRequest originalRequest = new SearchRequest()
+                .source(new SearchSourceBuilder()
+                        .fetchSource(false)
+                        .from(5)
+                        .size(10));
+        SearchRequest transformedRequest = ranker.preprocessRequest(originalRequest, configuration);
+        assertTrue(transformedRequest.source().fetchSource().fetchSource());
+        assertEquals(0, transformedRequest.source().from());
+        assertEquals(50, transformedRequest.source().size());
+    }
+
+    public void testShouldNotTransformWithoutSource() {
+        KendraIntelligentRanker ranker = new KendraIntelligentRanker(buildMockHttpClient());
+        SearchRequest originalRequest = new SearchRequest();
+        boolean shouldTransform = ranker.shouldTransform(originalRequest, new KendraIntelligentRankingConfiguration());
+        assertFalse(shouldTransform);
+    }
+
+    public void testShouldNotTransformWithoutQuery() {
+        KendraIntelligentRanker ranker = new KendraIntelligentRanker(buildMockHttpClient());
+        SearchRequest originalRequest = new SearchRequest();
```

Comment:
We should set the SearchSourceBuilder with query(null) here

Reply:
Oops! Thanks

I think I copy/pasted the test above, got distracted, then came back and started writing the next test. 😁